app/templates/settings.hbs

```diff
@@ -51,10 +51,14 @@
                       <a class="ui right floated disconnect button" {{action 'showConfirmDeleteModal' provider token}}>Disconnect</a>
                       
                       {{!-- TODO: Where do we information on who authorized the connection? --}}
-                      {{#if (and (eq provider.type 'apikey') token.resource_server)}}
+                      {{#if (and (and (eq provider.type 'apikey') token.resource_server) (eq provider.state 'authorized'))}}
```

Comment:
There is no `provider.state` when `provider.type==='apikey'`. Need to remove this clause.